components-api/src/main/scala/pl/touk/nussknacker/engine/api/process/OutputValidatorErrorsConverter.scala

```diff
@@ -0,0 +1,72 @@
+package pl.touk.nussknacker.engine.api.process
+
+import cats.data.NonEmptyList
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.CustomNodeError
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedUnion, TypingResult}
+
+object OutputValidatorErrorsConverter {
+
+  implicit class DisplayingTypingResult(typingResult: TypingResult) {
+    def displayType: String = typingResult match {
+      case un: TypedUnion if un.isEmptyUnion => "null" //Is it okey? Null is presented as EmptyUnion...
+      case _ => typingResult.display.capitalize
+    }
+  }
+
+}
+
+class OutputValidatorErrorsConverter(schemaParamName: String) {
+
+  import OutputValidatorErrorsMessageFormatter._
+
+  final def convertValidationErrors(errors: NonEmptyList[OutputValidatorError])(implicit nodeId: NodeId): CustomNodeError = {
+    val accumulator = (List.empty[OutputValidatorMissingFieldsError], List.empty[OutputValidatorRedundantFieldsError], List.empty[OutputValidatorTypeError])
+    val (missingFieldsError, redundantFieldsError, typeFieldsError) = errors.toList.foldRight(accumulator) {
```

Comment:
I wonder is it worthy to write this whole complex grouping logic instead of some simpler list...